### PR TITLE
fix: issue #1701 Null pointer Exception in models/Nutriments.java

### LINKED ISSUE #1701: java.lang.NullPointerException:    at openfoodfacts.github.scrachx.openfood.models.Nutriments.get
java.lang.NullPointerException: 
  at openfoodfacts.github.scrachx.openfood.models.Nutriments.get (Nutriments.java)
  at openfoodfacts.github.scrachx.openfood.views.product.nutrition_details.NutritionInfoProductFragment.access$000 (NutritionInfoProductFragment.java)
  or                     .getNutrimentItems (NutritionInfoProductFragment.java)
  or                     .lambda$onRequestPermissionsResult$0$NutritionInfoProductFragment (NutritionInfoProductFragment.java)
  or                     .onActivityResult (NutritionInfoProductFragment.java)
  or                     .onCreateView (NutritionInfoProductFragment.java)
  or                     .onPhotoReturned (NutritionInfoProductFragment.java)
  or                     .onRequestPermissionsResult (NutritionInfoProductFragment.java)
  or                     .onViewCreated (NutritionInfoProductFragment.java)
  or                     .refreshView (NutritionInfoProductFragment.java)
  at openfoodfacts.github.scrachx.openfood.views.product.nutrition_details.NutritionInfoProductFragment.access$000 (NutritionInfoProductFragment.java)
  or                     .getNutrimentItems (NutritionInfoProductFragment.java)
  or                     .lambda$onRequestPermissionsResult$0$NutritionInfoProductFragment (NutritionInfoProductFragment.java)
  or                     .onActivityResult (NutritionInfoProductFragment.java)
  or                     .onCreateView (NutritionInfoProductFragment.java)
  or                     .onPhotoReturned (NutritionInfoProductFragment.java)
  or                     .onRequestPermissionsResult (NutritionInfoProductFragment.java)
  or                     .onViewCreated (NutritionInfoProductFragment.java)
  or                     .refreshView (NutritionInfoProductFragment.java)
  at openfoodfacts.github.scrachx.openfood.views.product.nutrition_details.NutritionInfoProductFragment.access$000 (NutritionInfoProductFragment.java)
  or                     .getNutrimentItems (NutritionInfoProductFragment.java)
  or                     .lambda$onRequestPermissionsResult$0$NutritionInfoProductFragment (NutritionInfoProductFragment.java)
  or                     .onActivityResult (NutritionInfoProductFragment.java)
  or                     .onCreateView (NutritionInfoProductFragment.java)
  or                     .onPhotoReturned (NutritionInfoProductFragment.java)
  or                     .onRequestPermissionsResult (NutritionInfoProductFragment.java)
  or                     .onViewCreated (NutritionInfoProductFragment.java)
  or                     .refreshView (NutritionInfoProductFragment.java)
  at android.support.v4.app.FragmentManagerImpl.access$500 (FragmentManagerImpl.java)
  or                     .access$600 (FragmentManagerImpl.java)
  or                     .addFragment (FragmentManagerImpl.java)
  or                     .addOnBackStackChangedListener (FragmentManagerImpl.java)
  or                     .allocBackStackIndex (FragmentManagerImpl.java)
  or                     .animateRemoveFragment (FragmentManagerImpl.java)
  or                     .attachController (FragmentManagerImpl.java)
  or                     .beginTransaction (FragmentManagerImpl.java)
  or                     .completeExecute (FragmentManagerImpl.java)
  or                     .dispatchConfigurationChanged (FragmentManagerImpl.java)
  or                     .dispatchCreateOptionsMenu (FragmentManagerImpl.java)
  or                     .dispatchMultiWindowModeChanged (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentViewCreated (FragmentManagerImpl.java)
  or                     .dispatchOptionsItemSelected (FragmentManagerImpl.java)
  or                     .dispatchPrepareOptionsMenu (FragmentManagerImpl.java)
  or                     .dump (FragmentManagerImpl.java)
  or                     .enqueueAction (FragmentManagerImpl.java)
  or                     .executeOpsTogether (FragmentManagerImpl.java)
  or                     .executePostponedTransaction (FragmentManagerImpl.java)
  or                     .findFragmentById (FragmentManagerImpl.java)
  or                     .findFragmentByTag (FragmentManagerImpl.java)
  or                     .getAnimationListener (FragmentManagerImpl.java)
  or                     .getFragment (FragmentManagerImpl.java)
  or                     .loadAnimation (FragmentManagerImpl.java)
  or                     .makeFadeAnimation (FragmentManagerImpl.java)
  or                     .makeOpenCloseAnimation (FragmentManagerImpl.java)
  or                     .makeRemovedFragmentsInvisible (FragmentManagerImpl.java)
  or                     .modifiesAlpha (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .performPendingDeferredStart (FragmentManagerImpl.java)
  or                     .popBackStack (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .popBackStackState (FragmentManagerImpl.java)
  or                     .postponePostponableTransactions (FragmentManagerImpl.java)
  or                     .putFragment (FragmentManagerImpl.java)
  or                     .restoreAllState (FragmentManagerImpl.java)
  or                     .setBackStackIndex (FragmentManagerImpl.java)
  or                     .setRetaining (FragmentManagerImpl.java)
  or                     .shouldRunOnHWLayer (FragmentManagerImpl.java)
  or                     .throwException (FragmentManagerImpl.java)
  at android.support.v4.app.FragmentManagerImpl.dispatchOnFragmentStopped (FragmentManagerImpl.java)
  or                     .isStateSaved (FragmentManagerImpl.java)
  or                     .moveFragmentToExpectedState (FragmentManagerImpl.java)
  or                     .reverseTransit (FragmentManagerImpl.java)
  at android.support.v4.app.FragmentManagerImpl.access$500 (FragmentManagerImpl.java)
  or                     .access$600 (FragmentManagerImpl.java)
  or                     .addFragment (FragmentManagerImpl.java)
  or                     .addOnBackStackChangedListener (FragmentManagerImpl.java)
  or                     .allocBackStackIndex (FragmentManagerImpl.java)
  or                     .animateRemoveFragment (FragmentManagerImpl.java)
  or                     .attachController (FragmentManagerImpl.java)
  or                     .beginTransaction (FragmentManagerImpl.java)
  or                     .completeExecute (FragmentManagerImpl.java)
  or                     .dispatchConfigurationChanged (FragmentManagerImpl.java)
  or                     .dispatchCreateOptionsMenu (FragmentManagerImpl.java)
  or                     .dispatchMultiWindowModeChanged (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentViewCreated (FragmentManagerImpl.java)
  or                     .dispatchOptionsItemSelected (FragmentManagerImpl.java)
  or                     .dispatchPrepareOptionsMenu (FragmentManagerImpl.java)
  or                     .dump (FragmentManagerImpl.java)
  or                     .enqueueAction (FragmentManagerImpl.java)
  or                     .executeOpsTogether (FragmentManagerImpl.java)
  or                     .executePostponedTransaction (FragmentManagerImpl.java)
  or                     .findFragmentById (FragmentManagerImpl.java)
  or                     .findFragmentByTag (FragmentManagerImpl.java)
  or                     .getAnimationListener (FragmentManagerImpl.java)
  or                     .getFragment (FragmentManagerImpl.java)
  or                     .loadAnimation (FragmentManagerImpl.java)
  or                     .makeFadeAnimation (FragmentManagerImpl.java)
  or                     .makeOpenCloseAnimation (FragmentManagerImpl.java)
  or                     .makeRemovedFragmentsInvisible (FragmentManagerImpl.java)
  or                     .modifiesAlpha (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .performPendingDeferredStart (FragmentManagerImpl.java)
  or                     .popBackStack (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .popBackStackState (FragmentManagerImpl.java)
  or                     .postponePostponableTransactions (FragmentManagerImpl.java)
  or                     .putFragment (FragmentManagerImpl.java)
  or                     .restoreAllState (FragmentManagerImpl.java)
  or                     .setBackStackIndex (FragmentManagerImpl.java)
  or                     .setRetaining (FragmentManagerImpl.java)
  or                     .shouldRunOnHWLayer (FragmentManagerImpl.java)
  or                     .throwException (FragmentManagerImpl.java)
  at android.support.v4.app.BackStackRecord.executeOps (BackStackRecord.java)
  at android.support.v4.app.FragmentManagerImpl.addAddedFragments (FragmentManagerImpl.java)
  or                     .addBackStackState (FragmentManagerImpl.java)
  or                     .dispatchContextItemSelected (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentStarted (FragmentManagerImpl.java)
  or                     .dispatchOptionsMenuClosed (FragmentManagerImpl.java)
  or                     .dispatchPictureInPictureModeChanged (FragmentManagerImpl.java)
  or                     .execSingleAction (FragmentManagerImpl.java)
  or                     .executeOps (FragmentManagerImpl.java)
  or                     .findFragmentByWho (FragmentManagerImpl.java)
  or                     .getBackStackEntryAt (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .removeRedundantOperationsAndExecute (FragmentManagerImpl.java)
  or                     .setHWLayerAnimListenerIfAlpha (FragmentManagerImpl.java)
  or                     .transitToStyleIndex (FragmentManagerImpl.java)
  at android.support.v4.app.FragmentManagerImpl.access$500 (FragmentManagerImpl.java)
  or                     .access$600 (FragmentManagerImpl.java)
  or                     .addFragment (FragmentManagerImpl.java)
  or                     .addOnBackStackChangedListener (FragmentManagerImpl.java)
  or                     .allocBackStackIndex (FragmentManagerImpl.java)
  or                     .animateRemoveFragment (FragmentManagerImpl.java)
  or                     .attachController (FragmentManagerImpl.java)
  or                     .beginTransaction (FragmentManagerImpl.java)
  or                     .completeExecute (FragmentManagerImpl.java)
  or                     .dispatchConfigurationChanged (FragmentManagerImpl.java)
  or                     .dispatchCreateOptionsMenu (FragmentManagerImpl.java)
  or                     .dispatchMultiWindowModeChanged (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentPreCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentViewCreated (FragmentManagerImpl.java)
  or                     .dispatchOptionsItemSelected (FragmentManagerImpl.java)
  or                     .dispatchPrepareOptionsMenu (FragmentManagerImpl.java)
  or                     .dump (FragmentManagerImpl.java)
  or                     .enqueueAction (FragmentManagerImpl.java)
  or                     .executeOpsTogether (FragmentManagerImpl.java)
  or                     .executePostponedTransaction (FragmentManagerImpl.java)
  or                     .findFragmentById (FragmentManagerImpl.java)
  or                     .findFragmentByTag (FragmentManagerImpl.java)
  or                     .getAnimationListener (FragmentManagerImpl.java)
  or                     .getFragment (FragmentManagerImpl.java)
  or                     .loadAnimation (FragmentManagerImpl.java)
  or                     .makeFadeAnimation (FragmentManagerImpl.java)
  or                     .makeOpenCloseAnimation (FragmentManagerImpl.java)
  or                     .makeRemovedFragmentsInvisible (FragmentManagerImpl.java)
  or                     .modifiesAlpha (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .performPendingDeferredStart (FragmentManagerImpl.java)
  or                     .popBackStack (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .popBackStackState (FragmentManagerImpl.java)
  or                     .postponePostponableTransactions (FragmentManagerImpl.java)
  or                     .putFragment (FragmentManagerImpl.java)
  or                     .restoreAllState (FragmentManagerImpl.java)
  or                     .setBackStackIndex (FragmentManagerImpl.java)
  or                     .setRetaining (FragmentManagerImpl.java)
  or                     .shouldRunOnHWLayer (FragmentManagerImpl.java)
  or                     .throwException (FragmentManagerImpl.java)
  at android.support.v4.app.FragmentManagerImpl.addAddedFragments (FragmentManagerImpl.java)
  or                     .addBackStackState (FragmentManagerImpl.java)
  or                     .dispatchContextItemSelected (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentStarted (FragmentManagerImpl.java)
  or                     .dispatchOptionsMenuClosed (FragmentManagerImpl.java)
  or                     .dispatchPictureInPictureModeChanged (FragmentManagerImpl.java)
  or                     .execSingleAction (FragmentManagerImpl.java)
  or                     .executeOps (FragmentManagerImpl.java)
  or                     .findFragmentByWho (FragmentManagerImpl.java)
  or                     .getBackStackEntryAt (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .removeRedundantOperationsAndExecute (FragmentManagerImpl.java)
  or                     .setHWLayerAnimListenerIfAlpha (FragmentManagerImpl.java)
  or                     .transitToStyleIndex (FragmentManagerImpl.java)
  at android.support.v4.app.FragmentManagerImpl.addAddedFragments (FragmentManagerImpl.java)
  or                     .addBackStackState (FragmentManagerImpl.java)
  or                     .dispatchContextItemSelected (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentAttached (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentCreated (FragmentManagerImpl.java)
  or                     .dispatchOnFragmentStarted (FragmentManagerImpl.java)
  or                     .dispatchOptionsMenuClosed (FragmentManagerImpl.java)
  or                     .dispatchPictureInPictureModeChanged (FragmentManagerImpl.java)
  or                     .execSingleAction (FragmentManagerImpl.java)
  or                     .executeOps (FragmentManagerImpl.java)
  or                     .findFragmentByWho (FragmentManagerImpl.java)
  or                     .getBackStackEntryAt (FragmentManagerImpl.java)
  or                     .moveToState (FragmentManagerImpl.java)
  or                     .popBackStackImmediate (FragmentManagerImpl.java)
  or                     .removeRedundantOperationsAndExecute (FragmentManagerImpl.java)
  or                     .setHWLayerAnimListenerIfAlpha (FragmentManagerImpl.java)
  or                     .transitToStyleIndex (FragmentManagerImpl.java)
  at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss (BackStackRecord.java)
  at android.support.v4.app.FragmentPagerAdapter.finishUpdate (FragmentPagerAdapter.java)
  or                     .getItemId (FragmentPagerAdapter.java)
  or                     .setPrimaryItem (FragmentPagerAdapter.java)
  at android.support.v4.view.ViewPager.addNewItem (ViewPager.java)
  or                     .addOnAdapterChangeListener (ViewPager.java)
  or                     .addOnPageChangeListener (ViewPager.java)
  or                     .calculatePageOffsets (ViewPager.java)
  or                     .canScroll (ViewPager.java)
  or                     .completeScroll (ViewPager.java)
  or                     .determineTargetPage (ViewPager.java)
  or                     .distanceInfluenceForSnapDuration (ViewPager.java)
  or                     .executeKeyEvent (ViewPager.java)
  or                     .getChildRectInPagerCoordinates (ViewPager.java)
  or                     .infoForChild (ViewPager.java)
  or                     .initViewPager (ViewPager.java)
  or                     .isGutterDrag (ViewPager.java)
  or                     .onPageScrolled (ViewPager.java)
  or                     .onSecondaryPointerUp (ViewPager.java)
  or                     .populate (ViewPager.java)
  or                     .recomputeScrollPosition (ViewPager.java)
  or                     .scrollToItem (ViewPager.java)
  or                     .setCurrentItem (ViewPager.java)
  or                     .setCurrentItemInternal (ViewPager.java)
  or                     .smoothScrollTo (ViewPager.java)
  at android.support.v4.view.ViewPager.addNewItem (ViewPager.java)
  or                     .addOnAdapterChangeListener (ViewPager.java)
  or                     .addOnPageChangeListener (ViewPager.java)
  or                     .calculatePageOffsets (ViewPager.java)
  or                     .canScroll (ViewPager.java)
  or                     .completeScroll (ViewPager.java)
  or                     .determineTargetPage (ViewPager.java)
  or                     .distanceInfluenceForSnapDuration (ViewPager.java)
  or                     .executeKeyEvent (ViewPager.java)
  or                     .getChildRectInPagerCoordinates (ViewPager.java)
  or                     .infoForChild (ViewPager.java)
  or                     .initViewPager (ViewPager.java)
  or                     .isGutterDrag (ViewPager.java)
  or                     .onPageScrolled (ViewPager.java)
  or                     .onSecondaryPointerUp (ViewPager.java)
  or                     .populate (ViewPager.java)
  or                     .recomputeScrollPosition (ViewPager.java)
  or                     .scrollToItem (ViewPager.java)
  or                     .setCurrentItem (ViewPager.java)
  or                     .setCurrentItemInternal (ViewPager.java)
  or                     .smoothScrollTo (ViewPager.java)
  at android.support.v4.view.ViewPager.addNewItem (ViewPager.java)
  or                     .addOnAdapterChangeListener (ViewPager.java)
  or                     .addOnPageChangeListener (ViewPager.java)
  or                     .calculatePageOffsets (ViewPager.java)
  or                     .canScroll (ViewPager.java)
  or                     .completeScroll (ViewPager.java)
  or                     .determineTargetPage (ViewPager.java)
  or                     .distanceInfluenceForSnapDuration (ViewPager.java)
  or                     .executeKeyEvent (ViewPager.java)
  or                     .getChildRectInPagerCoordinates (ViewPager.java)
  or                     .infoForChild (ViewPager.java)
  or                     .initViewPager (ViewPager.java)
  or                     .isGutterDrag (ViewPager.java)
  or                     .onPageScrolled (ViewPager.java)
  or                     .onSecondaryPointerUp (ViewPager.java)
  or                     .populate (ViewPager.java)
  or                     .recomputeScrollPosition (ViewPager.java)
  or                     .scrollToItem (ViewPager.java)
  or                     .setCurrentItem (ViewPager.java)
  or                     .setCurrentItemInternal (ViewPager.java)
  or                     .smoothScrollTo (ViewPager.java)
  at android.support.v4.view.ViewPager.setCurrentItem (ViewPager.java)
  at android.support.design.widget.TabLayout$ViewPagerOnTabSelectedListener.onTabSelected (TabLayout.java)
  at android.support.design.widget.TabLayout.createLayoutParamsForTabs (TabLayout.java)
  or                     .dispatchTabSelected (TabLayout.java)
  at android.support.design.widget.TabLayout.dpToPx (TabLayout.java)
  or                     .removeAllTabs (TabLayout.java)
  or                     .removeOnTabSelectedListener (TabLayout.java)
  or                     .selectTab (TabLayout.java)
  at android.support.design.widget.TabLayout.dpToPx (TabLayout.java)
  or                     .removeAllTabs (TabLayout.java)
  or                     .removeOnTabSelectedListener (TabLayout.java)
  or                     .selectTab (TabLayout.java)
  at android.support.design.widget.TabLayout$Tab.select (TabLayout.java)
  at android.support.design.widget.TabLayout$TabView.performClick (TabLayout.java)
  at android.view.View$PerformClick.run (View.java:23645)
  at android.os.Handler.handleCallback (Handler.java:751)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6692)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)

### PR DESCRIPTION
**Pull request by paritosh-gupta**
## Description

Added checks for Null pointers in-case a particular data point was absent or present as null. It was trying to Convert a null to a string which was causing this issue. I believe there is a bigger root issue possibly here. Maybe the data should never be null however I think this check does not harm in either case.

## Related issues and discussion
Closes #1701 
 
 ## Screen-shots, if any
 
 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [ ] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Nutriments.java [hunk 1]
```diff
@@ -197,11 +197,16 @@
...
-        if (!additionalProperties.containsKey(nutrimentName)) {
+        if (nutrimentName.isEmpty() || !additionalProperties.containsKey(nutrimentName)) {
             return null;
         }
 
-        return new Nutriment(additionalProperties.get(nutrimentName).toString(), get100g(nutrimentName), getServing(nutrimentName), getUnit(nutrimentName));
+        try{
+            return new Nutriment(additionalProperties.get(nutrimentName).toString(), get100g(nutrimentName), getServing(nutrimentName), getUnit(nutrimentName));
+        }catch (NullPointerException e){
+        // In case one of the getters was unable to get data as string
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Nutriments.java]: You should print something to the log here for troubleshooting issues. Even `e.printStackTrace()` would be fine.

Comment (paritosh-gupta) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Nutriments.java]: Good idea 👍

**Comment by huzaifaiftikhar** [comment]
Build fails

screenshot_20180615-034842 2

**Comment by paritosh-gupta** [comment]
If this passes please squash and merge.

**Comment by teolemon** [comment]
@Karljoones

**Comment by teolemon** [comment]
@ross-holloway94 can we merge this one ?

**Pull request merged**